protocol/java/org/openyolo/protocol/AuthenticationDomain.java

```diff
@@ -137,21 +137,34 @@ public static AuthenticationDomain createAndroidAuthDomain(
 
     /**
      * Creates an authentication domain from its protocol buffer equivalent, in byte form.
-     * @throws IOException if the protocol buffer cannot be parsed.
+     * @throws MalformedDataException if the given protocol buffer is invalid.
      */
     @NonNull
     public static AuthenticationDomain fromProtobufBytes(@NonNull byte[] protobufBytes)
-            throws IOException {
-        return fromProtobuf(Protobufs.AuthenticationDomain.parseFrom(protobufBytes));
+            throws MalformedDataException {
+        validate(protobufBytes, notNullValue(), MalformedDataException.class);
+
+        try {
+            return fromProtobuf(Protobufs.AuthenticationDomain.parseFrom(protobufBytes));
+        } catch (IOException ex) {
+            throw new MalformedDataException("Unable to parse the given protocol buffer", ex);
+        }
     }
 
     /**
      * Creates an authentication domain from its protocol buffer equivalent.
+     * @throws MalformedDataException if the given protocol buffer is invalid.
      */
     @NonNull
     public static AuthenticationDomain fromProtobuf(
-            @NonNull Protobufs.AuthenticationDomain authDomain) {
-        return new AuthenticationDomain(authDomain.getUri());
+            @NonNull Protobufs.AuthenticationDomain authDomain) throws MalformedDataException {
+        validate(authDomain, notNullValue(), MalformedDataException.class);
+
+        try {
+            return new AuthenticationDomain(authDomain.getUri());
```

Comment:
AuthenticationDomain.fromProtobuf, directly throwing a MalformedDataException, would be better here.